### ISSUE TITLE
[Marketplace] Get multiple console errors when switching shops

### ISSUE THREAD

**Issue by zenweasel** [reactioncommerce/reaction#2750]
### Expected behavior
No errors

### Actual Behavior
Get a cascade of errors, probably all due to not waiting on the `Shop` subscription
```
TypeError: Cannot read property 'brandAssets' of undefined
    at Brand.getLogo (brand.js:23)
    at Brand.render (brand.js:33)
    at modules.js?hash=09e8e2dfa58d9bd3fd253c6052a7e955435c914c:107773
    at measureLifeCyclePerf (modules.js?hash=09e8e2dfa58d9bd3fd253c6052a7e955435c914c:107053)
    at ReactCompositeComponentWrapper._renderValidatedComponentWithoutOwnerOrContext (modules.js?hash=09e8e2dfa58d9bd3fd253c6052a7e955435c914c:107772)
    at ReactCompositeComponentWrapper._renderValidatedComponent (modules.js?hash=09e8e2dfa58d9bd3fd253c6052a7e955435c914c:107799)
    at ReactCompositeComponentWrapper._updateRenderedComponent (modules.js?hash=09e8e2dfa58d9bd3fd253c6052a7e955435c914c:107723)
    at ReactCompositeComponentWrapper._performComponentUpdate (modules.js?hash=09e8e2dfa58d9bd3fd253c6052a7e955435c914c:107701)
    at ReactCompositeComponentWrapper.updateComponent (modules.js?hash=09e8e2dfa58d9bd3fd253c6052a7e955435c914c:107622)
    at ReactCompositeComponentWrapper.receiveComponent (modules.js?hash=09e8e2dfa58d9bd3fd253c6052a7e955435c914c:107524)
```

### Steps to Reproduce the Behavior
Because this is a race-condition type thing you won't always see it, but usually after a couple of switches you can.

1. Login as admin
1. Create a second shop
1. As marketplace admin switch to second shop
1. Observe errors

### Versions
```
Node: 7.10.0
NPM: 5.3.0
Meteor Node: 4.8.4
Meteor NPM: 4.6.1
Reaction CLI: 0.11.0
Reaction: 1.5.0
Reaction branch: marketplace
Docker: 17.06.0-ce
```

**Comment by mikemurray** [comment]
@zenweasel @aanchirinah Made a PR here that should address this https://github.com/reactioncommerce/reaction/pull/2966

**Comment by impactmass** [comment]
closed by #2966
